src/FxCop/Desktop.Analyzers/Core/DesktopAnalyzersResources.resx

```diff
@@ -118,48 +118,36 @@
+    <value>Do Not Catch Corrupted State Exceptions</value>
   </data>
   <data name="DoNotCatchCorruptedStateExceptionsDescription" xml:space="preserve">
     <value>Do not author general catch handlers in code that receives corrupted state exceptions. Code that receives and intends to handle corrupted state exceptions should author distinct handlers for each exception type.</value>
   </data>
   <data name="DoNotCatchCorruptedStateExceptionsMessage" xml:space="preserve">
-    <value>Modify {0} to catch and handle a more specific set of exception type(s) than {1} or rethrow the exception. </value>
-  </data>
-  <data name="DoNotUseDES" xml:space="preserve">
-    <value>Do not use insecure cryptographic algorithm DES.</value>
-  </data>
-  <data name="DoNotUseDESDescription" xml:space="preserve">
-    <value>This type implements DES, a cryptographically insecure encryption algorithm. Replace this usage with an AES encryption algorithm (AES-256, AES-192 and AES-128 are acceptable) with a key length greater than or equal to 128 bits.</value>
-  </data>
-  <data name="DoNotUseMD5" xml:space="preserve">
-    <value>Do not use insecure cryptographic algorithm MD5.</value>
-  </data>
-  <data name="DoNotUseMD5Description" xml:space="preserve">
-    <value>This type implements MD5, a cryptographically insecure hashing function. Hash collisions are computationally feasible for the MD5 and HMACMD5 algorithms. Replace this usage with a SHA-2 family hash algorithm (SHA512, SHA384, SHA256).</value>
-  </data>
-  <data name="DoNotUseRC2" xml:space="preserve">
-    <value>Do not use insecure cryptographic algorithm RC2.</value>
-  </data>
-  <data name="DoNotUseRC2Description" xml:space="preserve">
-    <value>This type implements RC2, a cryptographically insecure encryption algorithm. Replace this usage with an AES encryption algorithm (AES-256, AES-192 and AES-128 are acceptable) with a key length greater than or equal to 128 bits.</value>
-  </data>
-  <data name="DoNotUseRIPEMD160" xml:space="preserve">
-    <value>Do not use insecure cryptographic algorithm RIPEMD160.</value>
-  </data>
-  <data name="DoNotUseRIPEMD160Description" xml:space="preserve">
-    <value>This type implements RIPEMD160, a cryptographically insecure hashing function. Hash collisions are computationally feasible for the RIPEMD hash algorithms. Replace this usage with a SHA-2 family hash algorithm (SHA512, SHA384, SHA256).</value>
+    <value>What: {0} is catching corrupted state exception.
+
+Why: This could mask errors (such as access violations), resulting in inconsistent state of execution or making it easier for attackers to compromise system.
+
+How: Modify {0} to catch and handle a more specific set of exception type(s) than {1} or re-throw the exception</value>
   </data>
-  <data name="DoNotUseDSA" xml:space="preserve">
-    <value>Do not use insecure cryptographic algorithm DSA.</value>
+  <data name="DoNotUseBrokenCryptographicAlgorithmsMessage" xml:space="preserve">
```

Comment:
Same comment as above:
Message should be - "{0} uses a broken cryptographic algorithm {1}."
Description - "An attack making it computationally feasible to break this algorithm exists. This allows attackers to break the cryptographic guarantees it is designed to provide. Depending on the type and application of this cryptographic algorithm, this may allow attackers to read enciphered messages, tamper with enciphered  messages, forge digital signatures, tamper with hashed content, or otherwise compromise any cryptosystem based on this algorithm. Replace encryption uses with the AES algorithm (AES-256, AES-192 and AES-128 are acceptable) with a key length greater than or equal to 128 bits. Replace hashing uses with a hashing function in the SHA-2 family, such as SHA512, SHA384, or SHA256. Replace digital signature uses with RSA with a key length greater than or equal to 2048-bits, or ECDSA with a key length greater than or equal to 256 bits.